### PR TITLE
Fix DSA indexer aiter (HIP) padding mismatch under DP-attention

### PR DESCRIPTION
**Pull request by dorado269**
## Motivation

On the HIP (aiter) DSA path, running Lightning-Indexer decode under
attention-TP / DP-attention crashes in the paged-MQA-logits top-k with

```text
Expected lengths.size(0) == B to be true
```

In `Indexer._get_topk_paged`, when attn-tp > 1 (or MAX_LEN padding) the hidden
states are padded to the largest per-rank token count, so `q_fp8`/`weights`
carry padding rows while `seqlens` (`dsa_seqlens_expanded`) is sized to the real
count `q_offset`. The CUDA backends (`deepgemm_paged_mqa_logits_split` /
`_native`, `cutedsl_paged_mqa_logits`) all take a `q_offset=` argument and slice
internally, so their `logits` is real-count-sized. The **aiter** branch passes
`q_fp8`/`weights` unsliced and sizes its `logits` output from `q_fp8.shape[0]`,
so the top-k then sees `score.shape[0] != lengths.shape[0]` and asserts.

The padding restore after `topk_transform` is additionally gated
`if not _is_hip`, so even once aiter is sliced HIP would not restore the padded
rows for the downstream consumer.

This is the same padding/length-mismatch class as #32762 (NPU, still open) — the
`aiter_paged_mqa_logits` path is its HIP sibling and is untouched on `main`. The
eager fallback that first exposes the seed-unavailable case landed in #30839.

## Modifications

`python/sglang/srt/layers/attention/dsa/dsa_indexer.py`,
`Indexer._get_topk_paged`:

- In the aiter branch, slice `q_fp8[:q_offset]` / `weights[:q_offset]` before
  `aiter_paged_mqa_logits`, matching the contract the CUDA backends already
  follow via `q_offset=`.
- Drop the `not _is_hip` guard on the padding restore. After the change every
  backend returns a result sized to the real `q_offset`, so the restore of the
  `-1` padding rows on the physical q length applies uniformly and no longer
  needs a platform guard.

Slicing is a no-op when there is no padding (`q_offset == q_fp8.shape[0]`); an
idle DP rank has `q_offset == 0` and passes the empty slice `q_fp8[:0]`, which
is what the CUDA path already does by slicing unconditionally.

## Accuracy Tests

Validated in a downstream deployment (GLM-5.2 MXFP4, MI355X/gfx950, sglang
0.5.15.post1) with DP-attention + EAGLE MTP: single-node dp8+ep8, EAGLE
steps=3 — 4/4 correctness, and conc=64 1k/1k **256/256 requests, zero failed**;
DP-attention-only and MTP-only paths unchanged. This PR is that fix rebased onto
current `main` (the surrounding dispatch has since gained the cutedsl / dg_native
branches). I don't have a CUDA DSA machine to re-run here — the CUDA branches are
untouched by design, and CI / a maintainer run would confirm on that base.

## Checklist

- [x] Format code with pre-commit
- [ ] Add unit tests
- [ ] Update documentation
- [ ] Provide accuracy and speed benchmark results

## Notes

Reported/related: #32762 (NPU sibling of this bug, open), #30839 (eager fallback
that surfaces it). Thanks to the DSA/indexer authors for the CUDA-side
`q_offset` slicing contract this follows.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #30982357295](https://github.com/sgl-project/sglang/actions/runs/30982357295)
Latest PR Test (Extra): :x: [Run #30982356918](https://github.com/sgl-project/sglang/actions/runs/30982356918)

### CODE REVIEW

**Comment by bingxche** [comment]
@amd-bot ci-status

**Comment by HaiShaw** [comment]
@bingxche @dorado269 can you can the asked test results from merge verdict?
